lib/runtime-core/src/typed_func.rs

```diff
@@ -438,33 +472,91 @@ macro_rules! impl_traits {
             }
         }
 
-        impl< $( $x, )* Rets, Trap, FN > ExternalFunction<( $( $x ),* ), Rets> for FN
+        impl< $( $x, )* Rets, Trap, FN > ExternalFunction<ExplicitVmCtx, ( $( $x ),* ), Rets> for FN
+        where
+            $( $x: WasmExternType, )*
+            Rets: WasmTypeList,
+            Trap: TrapEarly<Rets>,
+            FN: Fn(&mut vm::Ctx $( , $x )*) -> Trap,
+        {
+            #[allow(non_snake_case)]
+            fn to_raw(&self) -> NonNull<vm::Func> {
+                if mem::size_of::<Self>() == 0 {
```

Comment:
Presumably Rust wasn't happy with you using the `Sized` bound?

Reply:
`Self` refers to `FN` so to the `Fn` trait. Writing `mem::size_of::<Self>() == 0` is a way to tell: `FN` is a function pointer, or a closure with no captured environment.

This code will change in the next PR to support closures as host functions. So far, this PR only copy-paste the existing code by defining`vm::Ctx` as an optional argument.